### ISSUE TITLE
Connection Problem in Electron ( Or potentially any other browser )

### ISSUE THREAD

**Issue by eren23** [gram-js/gramjs#447]
Hello,

Yesterday all of a sudden, my app started to fail to connect/login Tegram after I close and reopen it, it's still like that for me, the console says; 
`WebSocket connection to 'ws://vesta.web.telegram.org/apiws' failed: Error during WebSocket handshake: Unexpected response code: 302`

and the error I see;
![error_mtp](https://user-images.githubusercontent.com/20106221/206396929-bc26063f-d6eb-4263-81f0-a902d9b30e7f.png)

As far as I understood from the older issues on similar topics the issue might relate to the ipv6 stuff, I tried toggling some flags during connection initialization ( useIPV6: false, useWSS: false, etc. ) and some other combinations.

Do you think that I can overcome this issue while still using a browser-side auth, or should I completely switch to a dedicated server for that, what would be the optimal solution in this case?

Note: My country is not blocked Telegram and I can access the interface using the node implementation, the node implementation tries to connect; 149.154.167.91:80/TCPFull (succeeds) 
while my electron app is trying to connect to; vesta.web.telegram.org/apiws (fails)

**Comment by painor** [comment]
can you maybe try using wss insetad of ws? ( set the use wss to true)

**Comment by eren23** [comment]
> 
Hello, thanks a lot for the reply.

I tried it again now, my current setting is; 
`{
            connectionRetries: 5,
            // useWSS: true,
            // useIPV6: false,
            useWSS: true,
            // testServers: false,
}`
And the error on the console is;
`browser.js?d017:25 WebSocket connection to 'ws://vesta.web.telegram.org/apiws' failed: Error during WebSocket handshake: Unexpected response code: 302`

**Comment by painor** [comment]
it doesn't seem like it worked. what's with the "//" in the code ?

Are you sure you're running the latest version of your code?  
when setting useWSS to true the library should connect to "wss://...." instead of "ws://..."

**Comment by eren23** [comment]
> it doesn't seem like it worked. what's with the "//" in the code ?
> 
> Are you sure you're running the latest version of your code? when setting useWSS to true the library should connect to "wss://...." instead of "ws://..."

Sorry for the confusion, //s are just commented lines but in new lines, somehow got copied here like that, it's actually            ```
 connectionRetries: 5,
            useWSS: true,
```

About the version, my console says, [2022-12-08T11:51:45.169] [INFO] - [Running gramJS version 2.12.7] and in my package.json it's "telegram": "^2.12.8"

**Comment by painor** [comment]
can you try updating to latest ? `npm i telegram@browser` (the package version should be 2.14.8)

**Comment by eren23** [comment]
Sorry, nevermind the last message it looks like it solved the issue with the update, let me check if everything is working and if so I'll close the issue, mega thanks for the help

**Comment by eren23** [comment]
> can you try updating to latest ? `npm i telegram@browser` (the package version should be 2.14.8)

Something weird came up now, when I try to use the login code I received an error prompt like this appears;
![weird_issue](https://user-images.githubusercontent.com/20106221/206406285-669afd0d-fdd2-445d-9987-322204677b1e.png)
Was not the case before.

**Comment by eren23** [comment]
> > can you try updating to latest ? `npm i telegram@browser` (the package version should be 2.14.8)
> 
> Something weird came up now, when I try to use the login code I received an error prompt like this appears; ![weird_issue](https://user-images.githubusercontent.com/20106221/206406285-669afd0d-fdd2-445d-9987-322204677b1e.png) Was not the case before.

The issue on the console is; 
[2022-12-08T12:16:04.626] [ERROR] - [Unhandled error while processing data]
Logger.js?c3fc:120 [2022-12-08T12:16:04.626] [ERROR] - [Error: Could not find a matching Constructor ID for the TLObject that was supposed to be        read with ID 2409088552. Most likely, a TLObject was trying to be read when         it should not be read. Remaining bytes: 264]

The weird part is when I submit the code I actually connect the telegram, my phone app says that, but I can't get my login string and other stuff in the app.

**Comment by eren23** [comment]
Okay here is how I solved it, I downgraded to the old version 2.12.7, then I placed useWss to both parts I connect a telegram client one in phone submitting ( for the cases when you don't have a string registered to the system ) and the other one is when you have your string already.

My mistake originally was not to put useWss flag when submitting a phone number, totally my bad but I think the issue about mtproto entity I highlighted above still exists in the latest version and is kind of irrelevant.

If you still want to reproduce the issue, I'm not able to share the whole project at the moment due to reasons out of my control but what I can do is share the project to bootstrap my project; https://github.com/saltyshiomix/nextron I think if there are packaging issues there I think this should help. 

I can also create a version with the failing settings and use it if you want me to test anything. ( In case of you not being able to reproduce)

Huge thanks for this great project and the great support with the issue.
